src/Platform/Microsoft.Testing.Platform/DotnetTest/IPC/DotnetTestDataConsumer.cs

```diff
@@ -33,8 +40,17 @@ public DotnetTestDataConsumer(NamedPipeClient dotnetTestPipeClient)
 
     public string Description => "Send back information to the dotnet test";
 
+    private string? ExecutionId => Environment.GetEnvironmentVariable(EnvironmentVariableConstants.TESTINGPLATFORM_DOTNETTEST_EXECUTIONID);
+
     public async Task ConsumeAsync(IDataProducer dataProducer, IData value, CancellationToken cancellationToken)
     {
+        if (DotnetTestConnection?.IsConnected == false)
```

Comment:
I suppose we can also avoid to register  the `DotnetTestDataConsumer` is it's not connected so we save computation.